lib/common_test/src/ct_console.erl

```diff
@@ -0,0 +1,174 @@
+    PaddingSizeRight = header_right_padding(Columns, PaddingSizeLeft, MessageSize),
+    {PaddingSizeLeft, PaddingSizeRight}.
+
+
+-spec pluralize(non_neg_integer(), string()) -> string().
+pluralize(Amount, Singular) ->
+    pluralize(Amount, Singular, Singular ++ "s").
+
+-spec pluralize(non_neg_integer(), string(), string()) -> string().
+pluralize(1, Singular, _Plural) -> Singular;
+pluralize(_, _Singular, Plural) -> Plural.
+
+
+-spec terminal_width(atom()) -> {ok, pos_integer()}.
+terminal_width(Driver) ->
+    case io:columns(Driver) of
+        {ok, _Columns} = Result ->
+            Result;
+        {error, enotsup} ->
+            {ok, 80}
+    end.
+
+-spec header_right_padding(pos_integer(), pos_integer(), pos_integer()) -> pos_integer().
+header_right_padding(_Columns, LeftPadding, MessageSize) when MessageSize rem 2 == 1 ->
+    LeftPadding - 1;
+header_right_padding(Columns, LeftPadding, _MessageSize) when Columns rem 2 == 0 ->
+    LeftPadding;
+header_right_padding(_Columns, LeftPadding, _MessageSize) ->
+    LeftPadding + 1.
+
+-spec size_on_terminal(iolist()) -> non_neg_integer().
+size_on_terminal(?TERM_BOLD ++ Rest) -> size_on_terminal(Rest);
+size_on_terminal(?TERM_BOLD_GREEN ++ Rest) -> size_on_terminal(Rest);
+size_on_terminal(?TERM_BLACK ++ Rest) -> size_on_terminal(Rest);
+size_on_terminal(?TERM_RED ++ Rest) -> size_on_terminal(Rest);
+size_on_terminal(?TERM_GREEN ++ Rest) -> size_on_terminal(Rest);
+size_on_terminal(?TERM_YELLOW ++ Rest) -> size_on_terminal(Rest);
+size_on_terminal(?TERM_CLEAR ++ Rest) -> size_on_terminal(Rest);
+size_on_terminal([Items | Rest]) when is_list(Items) -> size_on_terminal(Items) + size_on_terminal(Rest);
+size_on_terminal([Char | Rest]) when is_integer(Char) -> 1 + size_on_terminal(Rest);
+size_on_terminal([]) -> 0.
+
+%% @doc Elide framework code from the given traceback.
+-spec elide_framework_code(list()) -> list().
+elide_framework_code([{test_server, _Function, _Location} | Rest]) ->
+    elide_framework_code(Rest);
+elide_framework_code([Frame | Rest]) ->
+    [Frame | elide_framework_code(Rest)];
+elide_framework_code([]) ->
+    [].
+
+
+%% @doc Turn the given seconds into a human-readable string.
+-spec format_time(non_neg_integer()) -> string().
+format_time(Seconds) ->
+    Minutes = trunc(Seconds / 60),
+    RemainingSeconds = Seconds rem 60,
+    case Minutes of
+        0 ->
+            io_lib:format("~w ~s", [Seconds, pluralize(Seconds, "second")]);
+        _ ->
+            io_lib:format("~w ~s and ~w ~s", [Minutes, pluralize(Minutes, "minute"),
+                                              RemainingSeconds, pluralize(RemainingSeconds, "second")])
+    end.
+
+
+%% @doc
+%% Return the given string if colorized output is wanted, else an empty string.
+%% `mc' => "maybe color".
+%% @end
+-spec mc(string()) -> string().
+mc(Color) ->
+    case os:getenv("NO_COLOR") of
+        false ->
+            Color;
+        Value when Value =/= "" ->
```

Comment:
I see setting NO_COLOR to true disables colors in terminal output (kind of expected ;-))

But when redirected to text file, it still contains funny characters (file not being just text)
` ^[[0m22 passed^[[0m in 0 seconds^[[0m`
Is this wanted? Please explain.

Pure text files are nice for further processing ...